lib/devices/ios/ios.js

```diff
@@ -328,48 +328,30 @@ IOS.prototype.startInstruments = function (cb) {
     });
   }.bind(this);
 
-  if (this.shouldIgnoreInstrumentsExit()) {
-    // for safari on a real device, instruments is going to crash, but we
-    // expect that and ignore it
-    logger.debug("Not starting command proxy since we're expecting that " +
-                 "Instruments won't be able to talk to it anyway");
-    this.instruments.start(
-      function (err) {
-        if (err) return treatError(err, cb);
-        // instruments will never check into the command proxy, so go ahead
-        // and just call back here once instruments is started, rather than
-        // waiting for it to make the first connection to the command proxy
-        cb();
-      }
-    , function () {
-        // ignore unexpected instruments exit since that's exactly what
-        // we're expecting anyway
-    });
-  } else {
-    logger.debug("Starting command proxy.");
-    this.commandProxy.start(
-      function (err) { // on first connection
-        // first let instruments know so that it does not restart itself
-        this.instruments.launchHandler(err);
-        // then we call the callback
-        cb(err);
-      }.bind(this)
-      //
-    , function (err) { // regular cb
-        if (err) return treatError(err, cb);
-        logger.debug("Starting instruments");
-        this.instruments.start(
-          function (err) {
-            if (err) return treatError(err, cb);
-            // we don't call cb here, waiting for first connection or error
-          }.bind(this)
-        , function (code) {
+  logger.debug("Starting command proxy.");
+  this.commandProxy.start(
+    function (err) { // on first connection
```

Comment:
Something we can do to improve the code (while we are still using callback), is naming anonymous function when it is not obvious what they do. 
```
this.commandProxy.start(
   function onFirstConnection(err) { 
```